bridgectrl/interfaces.go

```diff
@@ -11,11 +11,15 @@ import (
 // merkleTreeStore interface for the Merkle Tree
 type merkleTreeStore interface {
 	Get(ctx context.Context, key []byte, dbTx pgx.Tx) ([][]byte, error)
-	Set(ctx context.Context, key []byte, value [][]byte, dbTx pgx.Tx) error
+	Set(ctx context.Context, key []byte, value [][]byte, rootID uint64, dbTx pgx.Tx) error
 	ResetMT(ctx context.Context, depositCount uint, network uint8, dbTx pgx.Tx) error
 	GetRoot(ctx context.Context, depositCount uint, network uint8, dbTx pgx.Tx) ([]byte, error)
-	SetRoot(ctx context.Context, root []byte, depositCount uint, network uint8, dbTx pgx.Tx) error
+	SetRoot(ctx context.Context, root []byte, depositCount uint, network uint8, dbTx pgx.Tx) (uint64, error)
 	GetLastDepositCount(ctx context.Context, network uint8, dbTx pgx.Tx) (uint, error)
+	// atomic
+	Rollback(ctx context.Context, dbTx pgx.Tx) error
+	BeginDBTransaction(ctx context.Context) (pgx.Tx, error)
+	Commit(ctx context.Context, dbTx pgx.Tx) error
```

Comment:
instead of creating the dbtx here, we could use the dbtx created in the synchronizer to when storing a deposit. WDYT?? Also we could do the same for reseting the mt

Reply:
that's great, I will research if there is any limit of `pgx.tx`, because I am feeling we are trying to execute massive queries in one time.